### ISSUE TITLE
Calculation mistake in `nCols` formula

### ISSUE THREAD

**Issue by mtpython** [janosh/svelte-bricks#30]
### Steps to reproduce

1. Open the [live demo](https://bricks.janosh.dev/), keep the default `masonryWidth = 1120px` and `gap = 20px`
2. Find the largest possible width $\text{columnWidth}$ of a single column in a three column layout:
    
    $$3 * \text{columnWidth} + 2 * \text{gap} = \text{masonryWidth}$$
    
    $$3 * \text{columnWidth} + 2 * 20px = 1120px$$
    
    $$\text{columnWidth} = 360px$$

3. Set `minColWidth` to `360px`

### Expected

There are 3 columns, each `360px` wide

### Actual

There are 2 columns 😱

And it switches to 2 columns at `minColWidth = 354px`

### Current formula

https://github.com/janosh/svelte-bricks/blob/1123d8a7b45e087f88d2a17406a5b54bd600962d/src/lib/Masonry.svelte#L28

$$\text{columnAmount} =  {\text{masonryWidth} \over {\text{columnWidth} + \text{gap}}}$$

### Solution

$$\text{columnAmount} * \text{columnWidth} + (\text{columnAmount} - 1) * \text{gap} = \text{masonryWidth}$$

$$\text{columnAmount} * \text{columnWidth} + \text{columnAmount} * \text{gap} = \text{masonryWidth}  + \text{gap}$$

$$\text{columnAmount} * (\text{columnWidth} + \text{gap}) = \text{masonryWidth}  + \text{gap}$$

$$\text{columnAmount} =  {{\text{masonryWidth} + \text{gap}} \over {\text{columnWidth} + \text{gap}}}$$

### TL;DR

The current formula is miscounting one `gap`, it should be changed like so:

```diff
- $: nCols = Math.min(items.length, Math.floor(masonryWidth / (minColWidth + gap)) || 1)
+ $: nCols = Math.min(items.length, Math.floor((masonryWidth + gap) / (minColWidth + gap)) || 1)
```

**Comment by janosh** [comment]
Thanks for the great bug report! 👍
